lib/rc.sh

```diff
@@ -19,15 +30,25 @@ source "$(dirname "${BASH_SOURCE[0]}")/pulumi.sh"
 add_artifacts() {
     local -r stack="${1}"
     local -r input_json="${2}"
-
-    jq -r 'to_entries | .[] | [.key, .value] | @tsv' <<< "${input_json}" |
-        while IFS=$'\t' read -r key value; do
-            pulumi config set \
-                --path "artifacts.${key}" \
-                "${value}" \
-                --cwd="$(project_directory "${stack}")" \
-                --stack="$(fully_qualified_stack_name "${stack}")"
-        done
+    local -r cwd="$(project_directory "${stack}")"
+    local -r stack_fq="$(fully_qualified_stack_name "${stack}")"
+
+    # Figure out which commands to send pulumi.
+    # requires pex be built
+    get_pulumi_commands="${ARTIFACTS_INTO_JSON_PEX} \
+        --input_json='${input_json}' \
+        --cwd='${cwd}' \
+        --stack='${stack_fq}' \
+    "
+    pulumi_commands_str=$(eval "${get_pulumi_commands}")
+    # grab \n-delimed commands into a bash array $pulumi_commands
+    readarray -t pulumi_commands <<< "${pulumi_commands_str}"
+
+    for pulumi_command in "${pulumi_commands[@]}"; do
+        # Note: `eval` would swallow useful quotes generated by the binary
+        # shellcheck disable=SC2091
+        $(${pulumi_command})
+    done
```

Comment:
the reason for all this "communicating bash commands across stdout and then eval'ing them"
is
I didn't want to port the shunit tests to python because lazy :)